doc/content/htgr/generic-pbr-tutorial/step1.md

```diff
@@ -0,0 +1,180 @@
+        caption=Geometry for Step 1.
+
+### Setting the Fluid Properties
+
+Fluid properties are set by first defining a `HeliumFluidProperties` fluid properties object
+and second by creating a material that inserts the fluid properties into functors (if you do not know what a functor is then click [here](https://mooseframework.inl.gov/moose/syntax/Materials/)).
+
+The first task is performed by the following input syntax:
+
+!listing htgr/generic-pbr-tutorial/step1.i block=FluidProperties
+
+The second task is performed by the `GeneralFunctorFluidProps` object:
+
+!listing htgr/generic-pbr-tutorial/step1.i block=Materials
+
+This object takes pressure and temperature provided as variables to parameters
+`pressure` and `T_fluid`, respectively, and computes fluid properties, such as
+density and viscosity. Density and viscosity are named `rho` and `mu`.
+Note that `T_fluid` is set to a constant value in this input file because
+we do not solve the heat equation and therefore have no fluid temperature defined
+as a variable.
+
+This object also requires providing functors to the `porosity`, `speed`, and `characteristic_length`
+parameters.
+
+Porosity is defined as an auxiliary variable here:
+
+!listing htgr/generic-pbr-tutorial/step1.i block=AuxVariables
+
+and set equal to the `bed_porosity` parameter of 0.39.
+`speed` is a functor that is computed automatically by the Navier-Stokes FV action,
+and `characteristic_length` is only used for computing Reynolds numbers, which are not
+required in Step 1; therefore, we set `characteristic_length = 1` for now.
+
+### Setting up the Thermal-Hydraulics Problem
+
+The thermal-hydraulics problem is set up using the finite volume Navier-Stokes action.
+This action is discussed in detail [here](https://mooseframework.inl.gov/source/actions/NSFVAction.html).
+The corresponding block in the input file is:
+
+!listing htgr/generic-pbr-tutorial/step1.i block=Modules
+
+Understanding the finite volume Navier-Stokes action is important.
+Therefore, each line will be explained.
+
+!listing htgr/generic-pbr-tutorial/step1.i start=compressibility end=porous_medium_treatment
+
+This line indicates that weakly compressible fluid equations are solved. These equations allow the density of the fluid to vary, and the discretization methods assume Mach numbers of less than 0.2.
+For compressible flows in reactors, the weakly compressible formulation is the most commonly used equation set.
+
+!listing htgr/generic-pbr-tutorial/step1.i start=porous_medium_treatment end=density
+
+This line activates the porous flow treatment. Porous flow indicates that fluid
+and solid share a volume and exchange mass, momentum, and energy in this volume.
+This exchange is usually described by correlations. In this input file, no mass, momentum, or energy exchange occurs. The porosity defines how
+much of each porous flow element is fluid. The porosity in this input file is defined
+as a variable and provided in the line:
```

Comment:
did you want to use the smoothing? this isnt compatible with variables for porosity in parallel , should be functor matprop